components/sidebar/index.js

```diff
@@ -3,11 +3,11 @@
 
 import {connect} from 'react-redux';
 import {bindActionCreators} from 'redux';
+import {createSelector} from 'reselect';
 
 import {Preferences} from 'mattermost-redux/constants/index';
 import {
     getCurrentChannel,
-    getUnreads,
```

Comment:
This returns "unreads" for those channels currently at store only.  For other teams that haven't visited yet, "unreads" for those channels are not considered.